janus_server/src/task.rs

```diff
@@ -253,43 +273,120 @@ impl Task {
         }
 
         // Compute hpke_configs mapping cfg.id -> (cfg, key).
-        let hpke_configs: HashMap<HpkeConfigId, (HpkeConfig, HpkePrivateKey)> = hpke_keys
+        let hpke_keys: HashMap<HpkeConfigId, (HpkeConfig, HpkePrivateKey)> = hpke_keys
             .into_iter()
             .map(|(cfg, key)| (*cfg.id(), (cfg, key)))
             .collect();
-        if hpke_configs.is_empty() {
+        if hpke_keys.is_empty() {
             return Err(Error::InvalidParameter("hpke_configs"));
         }
 
         Ok(Self {
-            id: task_id,
+            task_id,
             aggregator_endpoints,
+            query_type,
             vdaf,
             role,
             vdaf_verify_keys,
-            max_batch_lifetime,
+            max_batch_query_count,
+            task_expiration,
             min_batch_size,
-            min_batch_duration,
+            time_precision,
             tolerable_clock_skew,
             collector_hpke_config,
             aggregator_auth_tokens,
             collector_auth_tokens,
-            hpke_keys: hpke_configs,
+            hpke_keys,
         })
     }
 
+    /// Retrieves the task ID associated with this task.
+    pub fn task_id(&self) -> &TaskId {
```

Comment:
`task.task_id()` stutters. How about `Task::id(&self) -> &TaskId`? That would also be consistent with `HpkeConfig::id(&self)`.

Reply:
I went back and forth on this -- the reason I didn't go for it originally is that e.g. reports aren't actually identified just by `report_id`; their effective ID is actually `(task_id, report_id)`. So an `id()` method would return something like `(&TaskId, &ReportId)`.

Of course, that doesn't apply to tasks (since there is no higher-level entity tasks are conceptually "inside of"), and the stuttering bugs me too. Since you noticed it as well, I fixed it here as well as for ReportMetadata & AggregationJob. (The last ID type, BatchID, doesn't yet correspond directly to a defined data structure.)

Reply:
I went back and forth on this -- the reason I didn't go for it originally is that e.g. reports aren't actually identified just by `report_id`; their effective ID is actually `(task_id, report_id)`. So an `id()` method would return something like `(&TaskId, &ReportId)`.

Of course, that doesn't apply to tasks (since there is no higher-level entity tasks are conceptually "inside of"), and the stuttering bugs me too. Since you noticed it as well, I fixed it here as well as for ReportMetadata & AggregationJob. (The last ID type, BatchID, doesn't yet correspond directly to a defined data structure.)

Reply:
I went back and forth on this -- the reason I didn't go for it originally is that e.g. reports aren't actually identified just by `report_id`; their effective ID is actually `(task_id, report_id)`. So an `id()` method would return something like `(&TaskId, &ReportId)`.

Of course, that doesn't apply to tasks (since there is no higher-level entity tasks are conceptually "inside of"), and the stuttering bugs me too. Since you noticed it as well, I fixed it here as well as for ReportMetadata & AggregationJob. (The last ID type, BatchID, doesn't yet correspond directly to a defined data structure.)